AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2000,6 +2082,37 @@ - (void)layout
     subnode = ((ASDisplayNode *)subnodeLayout.layoutableObject);
     [subnode setFrame:subnodeFrame];
   }
+  
+  if ([[self class] usesImplicitHierarchyManagement]) {
+    if (!_managedSubnodes) {
+      _managedSubnodes = [NSMutableArray array];
```

Comment:
If this can be further deferred, we don't want every leaf node with no subnodes to create extra objects